src/hooks/useApplyUserDefinedFormStyling.ts

```diff
@@ -0,0 +1,203 @@
+import { StylingEngineTypes } from '@oneblink/types'
+import {
+  objectEntriesAsKeyValuePairs,
+  cssMappingsNeverKeyLog as neverKey,
+  CssMappings,
+} from '../services/styling-engine'
+import useApplyUserDefinedStyling from './useApplyUserDefinedStyling'
+
+const cssKeys = {
+  borderColour: 'border-color',
+  borderRadius: 'border-radius',
+  borderWidth: 'border-width',
+  padding: 'padding',
+  fontColor: 'color',
+  fontSize: 'font-size',
+  fontWeight: 'font-weight',
+  backgroundColour: 'background-color',
+  marginBottom: 'margin-bottom',
+} as const
+
+const stylingTargets = {
+  '.forms-renderer': {},
+  '.ob-form-container': {},
+  '.ob-element:not(:last-child)': {},
+  '.ob-label': {},
+  '.ob-heading': {},
+  '.title.is-1': {},
+  '.title.is-2': {},
+  '.title.is-3': {},
+  '.title.is-4': {},
+  '.title.is-5': {},
+} as const
+
+type HeadingSizeKey = 1 | 2 | 3 | 4 | 5
+
+const useApplyUserDefinedFormStyling = (
+  formStyling?: StylingEngineTypes.FormStyle,
+) => {
+  useApplyUserDefinedStyling(formStyling, applyCssMappings)
+}
+
+export default useApplyUserDefinedFormStyling
+
+const applyCssMappings = (formStyling: StylingEngineTypes.FormStyle) => {
+  return objectEntriesAsKeyValuePairs(formStyling).reduce<
+    CssMappings<typeof stylingTargets>
+  >((memo, { key, value }) => {
+    // if (keyVal.value === undefined) return memo
+    switch (key) {
+      case 'backgroundColour':
+        {
+          memo['.forms-renderer']['background-color'] = value
```

Comment:
I assume `forms-renderer` is defined in the forms renderer code base. If so, we prob don't want to use class names as selectors that do not exist in this code base.

Reply:
Yerp. forgot about this one.

Reply:
Need to talk to Blake about this one.

Reply:
Blake wants the outer background property (this one) to stay in the form section. So we need to either live with this, or we should move it to the app styles under the hood.